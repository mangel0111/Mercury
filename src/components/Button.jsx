import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ButtonBase = styled.button`
    display: inline-block;
    font-weight: 400;
    text-align: center;
    white-space: nowrap;
    vertical-align: middle;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
    border: 1px solid transparent;
    padding: .375rem .75rem;
    font-size: 1rem;
    line-height: 1.5;
    border-radius: .25rem;
    transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;    
    color: #fff;
    background-color: ${props => bgColorChooser(props.type)};
    border-color: ${props => bgColorChooser(props.type)};
`;

const bgColorChooser = (type) => {
	switch(type){
	case 'PRIMARY': 
		return  '#28a745';
	case 'SECONDARY':
		return '#6c757d';
	case 'DANGER': 
		return '#dc3545';
	case 'NEUTRAL':
	default: 
		return '#f8f9fa';
	}
};

class Button extends Component {
	render(){
		const { label, onClick, type = 'NEUTRAL'} = this.props;
		return (
			<ButtonBase
				type={type}
				onClick={onClick}
			>
				{label}
			</ButtonBase>
		);
	}
}

Button.propTypes = {
	label: PropTypes.string,
	onClick: PropTypes.func,
	type: PropTypes.string
};

export default Button;
