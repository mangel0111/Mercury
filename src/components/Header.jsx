import React, { Component } from 'react';
import styled from 'styled-components';

const HeaderNav = styled.header`
    padding: 0;
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 55px;
    z-index: 2;
    background-color: #000;
    color: #fff;
`;

const Container = styled.div`
    margin-right: 10%;
    margin-left: 10%;
    padding-left: 15px;
    padding-right: 15px;
`;

const Nav = styled.div`
    display: flex;
    margin: 10px;
    font-size: 20px;
    font-weight: lighter;
`;

const Menu = styled.div`
    margin-left: auto;
`;

export class Header extends Component{
	render (){
		return (
			<HeaderNav>
				<Container>
					<Nav>
						<div>
                            Mercury
						</div>
						<Menu>
                            Home
						</Menu>
					</Nav>
				</Container>
			</HeaderNav>
		);
	}
}

export default Header;
