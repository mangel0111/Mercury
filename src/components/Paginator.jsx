import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Navigator = styled.nav`
    display: flex;
`;

const PageNavigator =  styled.ul`
    margin-left: auto;
    display: flex;
    padding-left: 0;
    list-style: none;
    border-radius: .25rem;
    margin-top: .5rem;
    margin-bottom: .5rem;

    a {
        cursor: pointer;
    }
`;

const Link = styled.a`
    position: relative;
    display: block;
    padding: .5rem .75rem;
    margin-left: -1px;
    line-height: 1.25;
    color: #007bff;
    background-color: #fff;
    border: 1px solid #dee2e6;
    margin-left: 0;
`;

const FirtsLink = styled(Link)`
    border-top-left-radius: .25rem;
    border-bottom-left-radius: .25rem;
`;

const LastLink = styled(Link)`
    border-top-right-radius: .25rem;
    border-bottom-right-radius: .25rem;

`;

export class Paginator extends Component {
    
	getMidlePages() {
		const { pageSize, size } = this.props;
		let maxPages = Math.floor(size / pageSize);
		if(maxPages <= 1){
			return [];
		}
		const pages = Array.from(new Array(maxPages),(val,index)=>index);
		return pages.slice(1);
	}

	render(){
		const { pageSize, size, onChange } = this.props;
    
		if(!size || size <= pageSize) {
			return null;
		}
		return (
			<Navigator>
				<PageNavigator>
					<li>
						<FirtsLink
							onClick={()=> onChange({page: 0})}
						>
                            Previous
						</FirtsLink>
					</li>
					{this.getMidlePages().map(pageItem => 
						(
							<li 
								key={pageItem}
							>
								<Link
									onClick={()=> onChange({page: --pageItem})}s
								>
									{++pageItem}
								</Link>
							</li>
						)
					)}
					<li>
						<LastLink
							onClick={()=> onChange({page: Math.floor(size / pageSize)})}
						>
                            Last
						</LastLink>
					</li>
				</PageNavigator>
			</Navigator>
		);
	}
}

Paginator.propTypes = {
	onChange: PropTypes.func,
	size: PropTypes.number,
	pageSize: PropTypes.number,
	page: PropTypes.number
};

export default Paginator;
