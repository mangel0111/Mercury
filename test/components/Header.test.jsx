import React from 'react';
import renderer from 'react-test-renderer';
import { Header } from '../../src/components/Header';


test('Renders Header', () => {
	const rendered = renderer.create(<Header/>);
	const tree = rendered.toJSON();
	expect(tree).toMatchSnapshot();
});
