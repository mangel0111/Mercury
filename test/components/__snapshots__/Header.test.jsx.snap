// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Header 1`] = `
<header
  className="sc-bdVaJa jhfvel"
>
  <div
    className="sc-bwzfXH hWHZbd"
  >
    <div
      className="sc-htpNat cgSBXb"
    >
      <div>
        Mercury
      </div>
      <div
        className="sc-bxivhb fzMRlP"
      >
        Home
      </div>
    </div>
  </div>
</header>
`;
